src/status_im2/contexts/chat/lightbox/text_sheet/utils.cljs

```diff
@@ -8,23 +8,32 @@
 
 (defn sheet-gesture
   [{:keys [derived-value saved-top overlay-opacity gradient-opacity]}
-   expanded-height max-height overlay-z-index expanded? dragging?]
+   expanded-height max-height full-height overlay-z-index expanded? dragging? expanding-message?]
   (-> (gesture/gesture-pan)
+      (gesture/enabled expanding-message?)
       (gesture/on-start (fn []
                           (reset! overlay-z-index 1)
                           (reset! dragging? true)
-                          (reanimated/animate gradient-opacity 0)))
+                          (when (not expanded?)
+                            (reanimated/animate gradient-opacity 0))))
       (gesture/on-update
        (fn [e]
-         (let [new-value     (+ (reanimated/get-shared-value saved-top) (oops/oget e "translationY"))
-               bounded-value (max (min (- new-value) expanded-height) constants/text-min-height)
-               progress      (/ (- new-value) max-height)]
-           (reanimated/set-shared-value overlay-opacity progress)
-           (reanimated/set-shared-value derived-value bounded-value))))
+         (let [event-value     (oops/oget e "translationY")
+               old-value       (reanimated/get-shared-value saved-top)
+               new-value       (+ old-value event-value)
+               progress        (/ (- new-value) max-height)
+               upper-boundery? (< new-value (- full-height))
+               lower-boundary? (and (> new-value (- constants/text-min-height))
+                                    (pos? event-value))]
+           (when (and (not upper-boundery?) (not lower-boundary?))
+             (reanimated/set-shared-value overlay-opacity progress)
+             (reanimated/set-shared-value derived-value (- new-value))))))
       (gesture/on-end
        (fn []
-         (if (or (> (- (reanimated/get-shared-value derived-value))
-                    (reanimated/get-shared-value saved-top))
+         (if (or (and
+                  (not (> (reanimated/get-shared-value derived-value) max-height))
```

Comment:
`(reanimated/get-shared-value derived-value))` is called 5 times within `on-end`, it's better to extract to a local binding.